scripts/ASTImportTest.sh

```diff
@@ -1,105 +1,175 @@
+{
+    local error_message="$1"
+    local stderr_file="$2"
+    local stdout_file="$3"
+    printError "$error_message"
+    printError ""
+    printError "stderr:"
+    cat "$stderr_file" >&2
+    printError ""
+    printError "stdout:"
+    cat "$stdout_file" >&2
+}
+
+IMPORT_TEST_TYPE=
+EXIT_ON_ERROR=0
+for PARAM in "$@"
+do
+    case "$PARAM" in
+        ast) IMPORT_TEST_TYPE="ast" ;;
+        --exit-on-error) EXIT_ON_ERROR=1 ;;
+        *) print_usage ;;
+    esac
+done
 
 SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
 ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
-NSOURCES="$(find "$SYNTAXTESTS_DIR" -type f | wc -l)"
-
-# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
-# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
 
 FAILED=0
 UNCOMPILABLE=0
 TESTED=0
 
-if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
-    echo "Test directory not empty. Skipping!"
-    exit 1
-fi
+function ast_import_export_equivalence
+{
+    local sol_file="$1"
+    local input_files=( "${@:2}" )
+
+    local export_command=("$SOLC" --combined-json ast --pretty-json --json-indent 4 "${input_files[@]}")
+    local import_command=("$SOLC" --import-ast --combined-json ast --pretty-json --json-indent 4)
+
+    # export ast - save ast json as expected result (silently)
+    if ! "${export_command[@]}" > expected.json 2> stderr_export.txt
+    then
+        print_stderr_stdout "ERROR: AST export failed for input file $sol_file." ./stderr_export.txt ./expected.json
+        print_used_commands "$(pwd)" "${export_command[*]}" "${import_command[*]}"
+        return 1
+    fi
+
+    # (re)import ast - and export it again as obtained result (silently)
+    if ! "${import_command[@]}" expected.json > obtained.json 2> stderr_import.txt
+    then
+        print_stderr_stdout "ERROR: AST export failed for input file $sol_file." ./stderr_import.txt ./obtained.json
+        print_used_commands "$(pwd)" "${export_command[*]}" "${import_command[*]}"
+        return 1
+    fi
 
-# function tests whether exporting and importing again leaves the JSON ast unchanged
+    # compare expected and obtained ast's
+    if diff_files expected.json obtained.json
+    then
+        echo -n "✅"
+    else
+        printError "❌ ERROR: AST reimport failed for ${sol_file}"
+        if [[ $EXIT_ON_ERROR == 1 ]]
+        then
+            print_used_commands "$(pwd)" "${export_command[*]}" "${import_command[*]}"
+            return 1
+        fi
+        FAILED=$((FAILED + 1))
+    fi
+    TESTED=$((TESTED + 1))
+}
+
+# function tests whether exporting and importing again is equivalent.
 # Results are recorded by adding to FAILED or UNCOMPILABLE.
-# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Also, in case of a mismatch a diff is printed
 # Expected parameters:
 # $1 name of the file to be exported and imported
 # $2 any files needed to do so that might be in parent directories
 function testImportExportEquivalence {
```

Comment:
If all other functions in the file use underscores, I guess it would be best to rename this one too to use the same naming convention.